### ISSUE TITLE
importing playlists

### ISSUE THREAD

**Issue by 6arms1leg** [devsnd/cherrymusic#79]
there is already an issue for exporting playlists (#57). it would be so cool if there was also a feature of importing playlists. i know this would only be possible for playlists from music archives with the same full path, but that would already do the job (so that a user can re-import his/her exported playlists for example, or people that actually have the same music archive can exchange playlists).

EDIT: preferrably with an error message for alien playlists that have different paths and thus cant be added, so users wont be confused when importing a "wrong" playlist and nothing happens.

**Comment by frabjous** [comment]
It would *really really really* be nice to be able to import m3u playlists, so I don't have to manually redo the ones I have.

I tried converting them to JSON and then use devsnd's script. It appeared to work, but when I tried to load the playlist in the browser interface, it started to play, but then Firefox froze. This is perhaps because it was quite a long playlist with thousands of entries. Can cherrymusic not handle that? Or was it because the "titles" I used might not have matched exactly what it would have created for itself?

**Comment by devsnd** [comment]
hi!

if you have successfully imported the playlists and they started playing, then everything works as it should.

there are however problems with creating huge playlists... once you import such a playlist, CM will request the metadata for each and every track in the playlist. and CM will do it simultaneously. for each meta-info received, the whole playlist will rerender. so depending on the performance of your server you will either crash Firefox, because it cannot keep up with the rerendering of the playlist or you will DoS your server...

so in a nutshell: CM cannot handle this scenario right now.

there are two steps to resolve that issue: only rerender the playlist item that has received the metadata and only request metadata for one track at a time.

I think that both those steps have already been implemented for the media browser, so it shouldn't be too hard to get it working for the playlists as well. I'll try to do something about it soon, but I have very limited free time right now.

**Comment by frabjous** [comment]
That would be great... whenever you can get around to it. Thanks so much!
